### ISSUE TITLE
migrate from mdbook to docusaurus

### ISSUE THREAD

**Issue by zhangyangyu** [pingcap/tidb-dev-guide#166]
Currently the dev guide uses mdbook. But from my reading experience there are two problems: 1. the theme makes me feel bored and tired quickly, lacking color contrast; 2. no ToC (this is not supported by mdbook builtin). I'd like to propose [docusaurus](https://github.com/facebook/docusaurus) as the doc system. It could solve the two problems and it's actively maintained. There are already many sites hosted by it. 

A preliminary try is like: 
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/3690895/136898403-e06b4266-ebb6-4ab4-91b5-b5acd91de770.png">

But due to the lack of [category index page](https://github.com/facebook/docusaurus/issues/2643), we might need to wait several days still. Although I think the category pages of *contribute to TiDB* and *get started* are somehow redundant but the page of *understand TiDB* is valuable.

**Comment by zhangyangyu** [comment]
The try is a local site. But it uses the default theme of docusaurus so it should be nothing different from the [docusaurus tutorial](https://tutorial.docusaurus.io/docs/intro). Yes, too many doc/blog systems. Actually docusaurus gives [its subjective comparisons with other popular systems](https://docusaurus.io/docs#comparison-with-other-tools). docusaurus is not a must for me. I just think its out-of-box installation could solve mdbook's problems and it's actively maintained.
